gallery_dl/extractor/bilibili.py

```diff
@@ -81,6 +81,31 @@ def items(self):
             yield Message.Url, url, text.nameext_from_url(url, article)
 
 
+class BilibiliUserArticleFavlistExtractor(BilibiliExtractor):
+    subcategory = "user-article-favlist"
+    pattern = (r"(?:https?://)?(?:space.bilibili.com)"
+               r"/(\d+)/(?:favlist\?fid=opus)"
+               r"(?:&ftype=opus)?")
+    example = "https://space.bilibili.com/123456/favlist?fid=opus"
+    _warning = True
+
+    def _init(self):
+        BilibiliExtractor._init(self)
+        if self._warning:
+            if not self.cookies_check(("SESSDATA",)):
+                self.log.warning("no 'SESSDATA' cookie set")
+            BilibiliUserArticleFavlistExtractor._warning = False
+
+    def items(self):
+        mid = self.api.login_user_id()
+        if str(mid) != self.groups[0]:
+            raise exception.StopExtraction("This is not you favlist url!")
```

Comment:
I think doing an extra API request here is unnecessary. Just let the next API fail.